src/cdkutils/test_helpers/cloudwatch_alarm_simulator.py

```diff
@@ -0,0 +1,42 @@
+from typing import Any, Dict, List
+
+from aws_cdk import aws_cloudwatch
+
+ALARM_STATE_OK = "OK"
+ALARM_STATE_ALARM = "ALARM"
+
+
+class MetricException(Exception):
+    pass
+
+
+class CloudWatchAlarmSimulator:
+    def __init__(self, alarm_definition: aws_cloudwatch.AlarmProps):
+        self.alarm_definition = alarm_definition
+
+    def run_simulation(self, metrics: List[Dict[str, Any]]) -> Dict[int, str]:
+        for metric in metrics:
+            if metric["metric"] == self.alarm_definition.metric:
+                return {i: self.check_evaluation_period(i, metric["values"]) for i in range(len(metric["values"]))}
+        raise MetricException(f"Metric {self.alarm_definition.metric} not found!")
+
+    def check_evaluation_period(self, period: int, metric_values: List[float]) -> str:
+        evaluation_periods = int(self.alarm_definition.evaluation_periods)
+        metric_set = metric_values[max(1 + period - evaluation_periods, 0) : period + 1]
+        if all(self.check_breaching(metric) for metric in metric_set):
+            return ALARM_STATE_ALARM
+        return ALARM_STATE_OK
+
+    def check_breaching(self, metric: float) -> bool:
+        if self.alarm_definition.treat_missing_data == aws_cloudwatch.TreatMissingData.BREACHING and metric is None:
+            return True
+        if (
+            self.alarm_definition.comparison_operator == aws_cloudwatch.ComparisonOperator.LESS_THAN_THRESHOLD
+            and metric < self.alarm_definition.threshold
+        ):
+            return True
+        return False
```

Comment:
[LOW]

Reducing returning True if True else False to just return the bool expression

```suggestion
    def check_breaching(self, metric: float) -> bool:
        if self.alarm_definition.treat_missing_data == aws_cloudwatch.TreatMissingData.BREACHING and metric is None:
            return True
        return self.alarm_definition.comparison_operator == aws_cloudwatch.ComparisonOperator.LESS_THAN_THRESHOLD
            and metric < self.alarm_definition.threshold
```

Reply:
There will be a ton of other 'rules' to go by in here (for the other comparison operators etc.), so it won't be a straightforward if else once those are added - I'm currently thinking about how best to do this, but suggest leaving any refactoring here until the rest of the code is added?

Reply:
Sure

Reply:
This has been refactored now with a few additional rules